### ISSUE TITLE
Support for LIFX Light Bulbs?

### ISSUE THREAD

**Issue by HawkingJan** [philip1986/pimatic-led-light#68]
Hi,
I'm thinking about buying a LIFX Light Bulb
https://www.lifx.com
Even compared to the latest 3rd generation of Philips Hue they got good ratings, see e.g. the latest report here https://www.cnet.com/news/smart-home-showdown-philips-hue-vs-lifx/ 
I especially like the fact, that I can start with just one bulb and need no extra bridge and the LIFX brightness is larger.

Unfortunately I didn't find any pimatic plugin that supports LIFX.
Could support for this LIFX  device be easily included in this plugin? The official LIFX LAN API is provided here 
https://lan.developer.lifx.com
and a Python implementation can be found here
https://github.com/mclarkk/lifxlan
and a Node.js implementation 
https://github.com/MariusRumpf/node-lifx

Unfortunately I've no skills in programming CoffeeScript but maybe it is an easy extension that somebody can do. This would be highly appreciated, already many thanks!

**Comment by mwittig** [comment]
On the downside the LIFX bulbs are very expensive. Did you consider Yeelight (WiFi) as an alternative? Background: I am planning to provide a driver for yeelight, but currently there are no plans for LIFX unfortunately.

http://www.pcadvisor.co.uk/test-centre/digital-home/best-smart-light-2017-uk-best-smart-bulbs-you-can-control-from-your-phone-3647226/

**Comment by HawkingJan** [comment]
I agree that the LIFX bulbs are expensive (especially if compared to non-smart bulbs), but not that much if compared to the (also expensive but quite popular) Philips Hue.
As I said I want to start with just one coloured bulb (actually most lights in my apartment are based on the smaller E14 socket and to my knowledge there are no smart and bright bulbs with this socket available yet and I don't want to buy dozens of ugly adapters).  So for me the decision (based on prices from amazon.com) was between:
1 LIFX bulb 60 $ vs. Philips Hue Starter Kit 200 $ (OK, it contains 3 bulbs and the bridge, therefore the comparison is not completely fair).
But even if you already have the Hue bridge and want to equip your house with many more extra bulbs the prices are comparable:
Single Philips Hue bulb 50 $ vs. Pack of **4** LIFX bulbs 196 $

Regarding the Yeelight (WiFi) you mentioned, I also read about them before as they are not as expensive, however they are also significantly less bright:
Yeelight (WiFi) 30 $ for 600 lm
Philips Hue 50 $ for 800 lm
LIFX 60 $ for 1100 lm
Therefore the Yeelight are not really an option for me (doubling the sockets is not an option :-)).

Can you estimate how difficult it is to include LIFX into pimatic led-light (given the fact that the official API and an unofficial JS-libary are available)? Maybe I could convince you or somebody else to provide a driver for LIFX in the future...

**Comment by mwittig** [comment]
Well, give me some time to review. I found a page providing a "virtual bulb" - https://virtualbulb.lifx.co/ and some other projects alike. This makes it possible to test stuff without having the hardware in place. The library  node-lifx also looks nice and easy to use.

**Comment by amiraoul** [comment]
Any news on the lifx integration?
